test/models/patient_notification_eligibilty_test.rb

```diff
@@ -0,0 +1,400 @@
+      { last_date_of_exposure: 11.days.ago, created_at: 20.days.ago }
+    ].each do |workflow_params|
+      dependent = create(
+        :patient,
+        {
+          responder: patient,
+          monitoring: true,
+          closed_at: nil
+        }.merge(workflow_params)
+      )
+      assert_eligibility(patient, exp_eligibility)
+      dependent.destroy
+    end
+  end
+
+  def past_monitoring_period_dependent_test(patient, exp_eligibility: false)
+    [
+      { last_date_of_exposure: nil, created_at: 15.days.ago },
+      { last_date_of_exposure: 15.days.ago, created_at: 15.days.ago },
+      { last_date_of_exposure: 15.days.ago, created_at: 30.days.ago }
+    ].each do |workflow_params|
+      dependent = create(
+        :patient,
+        {
+          responder: patient,
+          monitoring: true,
+          closed_at: nil
+        }.merge(workflow_params)
+      )
+      assert_eligibility(patient, exp_eligibility)
+      dependent.destroy
+    end
+  end
+
+  def ignored_dependent_eligibility_fields_test(patient, exp_eligibility: true)
+    [
+      'E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call',
+      'SMS Text-message', 'Unknown', 'Opt-out', '', nil
+    ].each do |report_method|
+      [
+        { pause_notifications: true },
+        { last_assessment_reminder_sent: 1.hour.ago }
+      ].each do |workflow_params|
+        dependent = create(
+          :patient,
+          {
+            preferred_contact_method: report_method,
+            responder: patient,
+            monitoring: true,
+            closed_at: nil
+          }.merge(workflow_params)
+        )
+        assert_eligibility(patient, exp_eligibility)
+        # Creating an assessment from today SHOULD NOT affect eligibility
+        create(
+          :assessment,
+          patient: dependent,
+          symptomatic: false,
+          created_at: Time.now.getlocal('-04:00').beginning_of_day
+        )
+        assert_eligibility(patient, exp_eligibility)
+        dependent.destroy
+      end
+    end
+  end
+
+  test 'HoH notification eligible because of dependent flows' do
+    ['E-mailed Web Link', 'SMS Texted Weblink', 'Telephone call', 'SMS Text-message'].each do |report_method|
+      [
+        { isolation: true },
+        { continuous_exposure: true },
+        { last_date_of_exposure: nil, created_at: 5.days.ago },
+        { last_date_of_exposure: 5.days.ago, created_at: 5.days.ago },
+        { last_date_of_exposure: 11.days.ago, created_at: 20.days.ago }
+      ].each do |workflow_params|
+        [
+          { monitoring: false, closed_at: 1.day.ago },
+          { last_assessment_reminder_sent: 11.hours.ago },
+          { isolation: false, continuous_exposure: false, last_date_of_exposure: nil, created_at: 15.days.ago },
+          { isolation: false, continuous_exposure: false, last_date_of_exposure: 15.days.ago, created_at: 20.days.ago }
+        ].each do |ineligible_params|
+          Patient.destroy_all
```

Comment:
These tests call `destroy_all` a lot. Why is this required at the beginning of the tests? Fixtures being loaded?

Reply:
Further research may show that `Patient.destroy_all` would be causing some slowdowns with removing and having to re-load fixtures each time. Maybe worth looking in to an `assert_difference` approach.

Reply:
all destroy_all calls have been removed, and now just asserting that the HoH/non-household patient is eligible and that dependents are always ineligible. We also discussed how we can't really avoid doing individual destroys for dependents in the middle of tests but this doesn't slow it down nearly as much. Tests are currently broken right now but before I broke them this test file would complete in < 10 seconds